test/hapi_server.tests.js

```diff
@@ -7,118 +7,270 @@ const eventLogger = new EventLogger(bunyan.createLogger({
   name: 'test'
 }));
 
-var hapi_plugin = {
-  register: function(server, options, next) {
-    eventLogger.watch(server, { ignorePaths: ['/ignored'] });
-    next();
-  }
-};
+const startServer = (eventLoggerOptions, cb) => {
+  var hapi_plugin = {
+    register: function (server, options, next) {
+      eventLogger.watch(server, eventLoggerOptions);
+      next();
+    }
+  };
+
+  hapi_plugin.register.attributes = {
+    name: 'bunyan-logger',
+    version: '1.0.0'
+  };
 
-hapi_plugin.register.attributes = {
-  name: 'bunyan-logger',
-  version: '1.0.0'
+  const server = new Hapi.Server();
+  server.connection({ host: 'localhost', port: 9876 });
+  server.route({
+    method: 'GET',
+    path: '/',
+    handler: function (request, reply) {
+      return reply('Hello world!');
+    }
+  });
+  server.route({
+    method: 'GET',
+    path: '/ignored',
+    handler: function (request, reply) {
+      return reply('ignored!');
+    }
+  });
+  server.route({
+    method: 'GET',
+    path: '/slow',
+    handler: function (request, reply) {
+      setTimeout(function () {
+        return reply('Hellooooooo sloooooow woooooorld!');
+      }, 1500);
+    }
+  });
+  server.route({
+    method: 'POST',
+    path: '/internal_error',
+    handler: function (request, reply) {
+      return reply(new Error('test error'));
+    }
+  });
+  server.start(cb);
+  server.register(hapi_plugin, function (err) {
+    if (err) {
+      console.log('Failed to load Hapi plugin');
+    }
+  });
+  return server;
 };
 
 describe('watch Hapi server < v17', function () {
+  var loggerSave = {};
+
   var server;
 
-  before(function(done) {
-    server = new Hapi.Server();
-    server.connection({ host:'localhost', port: 9876 });
-    server.route({
-      method: 'GET',
-      path: '/',
-      handler: function(request, reply) {
-        return reply('Hello world!');
-      }
-    });
-    server.route({
-      method: 'GET',
-      path: '/ignored',
-      handler: function(request, reply) {
-        return reply('ignored!');
-      }
-    });
-    server.route({
-      method: 'GET',
-      path: '/slow',
-      handler: function(request, reply) {
-        setTimeout(function() {
-          return reply('Hellooooooo sloooooow woooooorld!');
-        }, 1500);
-      }
-    });
-    server.start(done);
-    server.register(hapi_plugin, function(err) {
-      if (err) {
-        console.log('Failed to load Hapi plugin');
-      }
-    });
+  before(function() {
+    loggerSave.info = eventLogger.logger.info;
+    loggerSave.error = eventLogger.logger.error;
   });
-
-  after(function (done) {
-    server.stop(done);
+  afterEach(function() {
+    eventLogger.logger.info = function() {};
+    eventLogger.logger.error = function() {};
+  });
+  after(function() {
+    // Restore saved logger functions to avoid test cross-contamination
+    eventLogger.logger.info = loggerSave.info;
+    eventLogger.logger.error = loggerSave.error;
```

Comment:
`server.stop()`?

Reply:
I need to stop the server in between scenarios below, so I don't think I can pull it higher here (it would only be called once at the end of the test suite).